docs/devel/contributing.md

```diff
@@ -369,6 +369,23 @@ to the commit message.
 Signed-off-by: Joe Smith <joe.smith@email.com>
 ```
 
+### Commit History Guidelines
+
+When authoring a PR, ensure that the commit history is clean and
+well-structured. Each commit should:
+
+- Contain a clear and concise title summarizing the changes.
+- Include a detailed description explaining the purpose and context of the
+  changes.
```

Comment:
Is this always required? Even for small commits?

Reply:
Good point, I reworded it.